proposals/bound-callable-references.md

```diff
@@ -84,6 +124,42 @@ fun test(c: C?) {
 }
 ```
 
+#### Nullable references
+
+According to this proposal, in the following syntax
+```
+Foo?::bar
+```
+the left-hand side can only be interpreted as a type because `Foo?` is not a valid expression. Therefore this would necessarily be a reference to an extension function named `bar` to the type `Foo?`.
```

Comment:
I'll answer anyway: if such expression is introduced, we'll have to prefer nullable types over such expressions in the LHS of `::`. I don't see what else we can do to support both nullable types and such expressions.